msim/src/sim/time/mod.rs

```diff
@@ -309,14 +322,33 @@ pub(crate) fn ensure_clocks() {
     }
 }
 
-/*
- * TODO: RocksDB schedules timers on multiple threads, including on the simulation thread, and it
- * can't handle having different views of the clock in different threads.
- * This should be fixable by creating the DBs in different threads, but that is somewhat invasive.
+fn use_real_wallcock() -> bool {
+    thread_local! {
+        // Note: setting MSIM_USE_REAL_WALLCLOCK will result in non-determinism for code that reads
+        // and acts on the absolute wall time. (Anything involving deltas only will still be
+        // deterministic).
+        static USE_REAL_WALLCLOCK: bool = {
```

Comment:
Not sure I understand this block.. why do we need the static var here when we have the envvar?

Reply:
for performance - we want to cache the env var lookup in a thread local since timing functions can be called very often.